src/rocprof_compute_soc/soc_base.py

```diff
@@ -186,12 +186,72 @@ def populate_mspec(self):
         self._mspec.gpu_model = mi_gpu_specs.get_gpu_model(
             self._mspec.gpu_arch, self._mspec.gpu_chip_id
         )
+
+        if not self._mspec.gpu_model:
+            self._mspec.gpu_model = self.detect_gpu_model(self._mspec.gpu_arch)
+
         self._mspec.num_xcd = str(
             mi_gpu_specs.get_num_xcds(
-                self._mspec.gpu_model, self._mspec.compute_partition
+                self._mspec.gpu_arch, self._mspec.gpu_model, self._mspec.compute_partition
             )
         )
 
+    @demarcate
+    def detect_gpu_model(self, gpu_arch):
+        """
+        Detects the GPU model using various identifiers from 'amd-smi static'.
+        Falls back through multiple methods if the primary method fails.
+        """
+
+        from utils.specs import run, search
+
+        amd_smi_static = run(["amd-smi", "static", "--gpu=0"], exit_on_error=True)
+
+        # Purposely search for patterns without variants suffix to try and match a known GPU model.
+        detection_methods = [
```

Comment:
Maybe add a comment here: we should check those through amd-smi python api in the future.